src/shipit_code_coverage/shipit_code_coverage/report_generators.py

```diff
@@ -1,54 +1,189 @@
 
+logger = get_logger(__name__)
+
+
+class ZeroCov(object):
+
+    DATE_FORMAT = '%Y-%m-%d %H:%M:%S %z'
+
+    def __init__(self, mc_repo_dir, hgmo_port=8000):
+        self.mc_repo_dir = mc_repo_dir
+        self.hgmo_port = str(hgmo_port)
+
+    def get_pid_file(self):
+        return '/tmp/hgmo.pid'
+
+    def get_pid(self):
+        with open(self.get_pid_file(), 'r') as In:
+            pid = In.read()
+            return int(pid)
+        return -1
+
+    def get_file_size(self, filename):
+        if self.mc_repo_dir:
+            filename = os.path.join(self.mc_repo_dir, filename)
+            if os.path.isfile(filename):
+                return os.path.getsize(filename)
+        return 0
+
+    def get_utc_from_timestamp(self, ts):
+        d = datetime.utcfromtimestamp(ts)
+        return d.replace(tzinfo=pytz.utc)
+
+    def get_date_str(self, d):
+        return d.strftime(ZeroCov.DATE_FORMAT)
+
+    def kill_hgmo(self):
+        pid = self.get_pid()
+        os.killpg(os.getpgid(pid), signal.SIGTERM)
+
+    def run_hgmo(self):
+        if not os.path.isdir(self.mc_repo_dir):
+            logger.warning('Not a directory for m-c', dir=self.mc_repo_dir)
+            return False
+
+        oldcwd = os.getcwd()
+        os.chdir(self.mc_repo_dir)
+        proc = subprocess.Popen(['hg', 'serve',
+                                 '--hgmo',
+                                 '--port', self.hgmo_port,
+                                 '--daemon',
+                                 '--pid-file', self.get_pid_file()])
+        proc.wait()
+        os.chdir(oldcwd)
+
+        pid = self.get_pid()
+        if pid == -1:
+            logger.error('hgmo is not running')
+            return False
+
+        logger.info('hgmo is running', pid=pid)
+        return True
 
-def zero_coverage(artifacts, out_dir='code-coverage-reports'):
-    report = grcov.report(artifacts, out_format='coveralls+')
-    report = json.loads(report.decode('utf-8'))  # Decoding is only necessary until Python 3.6.
-
-    zero_coverage_files = set()
-    zero_coverage_functions = {}
-    for sf in report['source_files']:
-        name = sf['name']
-
-        # For C/C++ source files, we can consider a file as being uncovered
-        # when all its source lines are uncovered.
-        all_lines_uncovered = all(c is None or c == 0 for c in sf['coverage'])
-        # For JavaScript files, we can't do the same, as the top-level is always
-        # executed, even if it just contains declarations. So, we need to check if
-        # all its functions, except the top-level, are uncovered.
-        all_functions_uncovered = True
-        for f in sf['functions']:
-            f_name = f['name']
-            if f_name == 'top-level':
+    def get_pushlog(self):
+        has_hgmo = self.run_hgmo()
+        if has_hgmo:
+            url = 'http://localhost:{}/json-pushes'.format(self.hgmo_port)
+        else:
+            url = 'https://hg.mozilla.org/mozilla-central/json-pushes'
```

Comment:
Since we don't need this anymore for tests, let's remove it altogether and always use the local URL